modin/pandas/test/dataframe/test_indexing.py

```diff
@@ -41,6 +41,40 @@
 matplotlib.use("Agg")
 
 
+def test_asof():
```

Comment:
Do you think it's necessary to add some testing to test some more common cases (no `NaN` values in the data) as well? And cases with >100 rows?

Reply:
Indeed. And, actually, I was thinking that adding Hypothesis-based testing would be a good idea, since it means not having to come up with all the test cases myself. So maybe this is an opportune time to do that.

Reply:
After a bit more thought: hypothesis will definitely improve testing coverage. However, it will also increase test suite runtime. I think that's a worthwhile tradeoff, the (semantic) coverage improvement will be significant, but for now I'll hold off on hypothesis and just add more manual/parametrized tests, and you tell me whether you're OK with that tradeoff.

In case you haven't encountered it: https://hypothesis.readthedocs.io/en/latest/